### PR TITLE
add github actions for gradle based builds

### PR DESCRIPTION
**Pull request by atomfrede**
updates #1297

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/ci/github/actions/application/GitHubActionsApplicationService.java [hunk 1]
```diff
@@ -14,6 +14,10 @@ public GitHubActionsApplicationService(GitHubActionsService gitHubActionsService
   }
 
   public void addGitHubActionsForMaven(Project project) {
-    gitHubActionsService.addGitHubActionsForMaven(project);
+    gitHubActionsService.addGithubActions(project);
+  }
+
+  public void addGithubActionsForGradle(Project project) {
+    gitHubActionsService.addGithubActions(project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/ci/github/actions/application/GitHubActionsApplicationService.java:21]: I think we can merge these 2 methods

File: src/main/java/tech/jhipster/lite/generator/ci/github/actions/domain/GitHubActionsService.java [hunk 2]
```diff
@@ -3,5 +3,5 @@
 import tech.jhipster.lite.generator.project.domain.Project;
 
 public interface GitHubActionsService {
-  void addGitHubActionsForMaven(Project project);
+  void addGithubActions(Project project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/ci/github/actions/domain/GitHubActionsService.java:6]: typo: addGithubActions -> addGitHubActions :-)

File: src/main/java/tech/jhipster/lite/generator/ci/github/actions/infrastructure/primary/rest/GitHubActionsResource.java [hunk 3]
```diff
@@ -24,12 +24,21 @@ class GitHubActionsResource {
...
   }
+
+  @Operation(summary = "Add GitHub Actions for Gradle Build")
+  @ApiResponse(responseCode = "500", description = "An error occurred while adding GitHub Actions")
+  @PostMapping("/gradle")
+  @GeneratorStep(id = GeneratorAction.GITHUB_ACTIONS)
+  public void addGithubActionsForGradle(@RequestBody ProjectDTO projectDTO) {
+    Project project = ProjectDTO.toProject(projectDTO);
+    gitHubActionsApplicationService.addGithubActionsForGradle(project);
+  }
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/ci/github/actions/infrastructure/primary/rest/GitHubActionsResource.java:43]: maybe merge both API, to use 1 single as the logic is in domainService

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/main/java/tech/jhipster/lite/generator/ci/github/actions/infrastructure/primary/rest/GitHubActionsResource.java (hunk 3 above)
Comment (atomfrede) [reply src/main/java/tech/jhipster/lite/generator/ci/github/actions/infrastructure/primary/rest/GitHubActionsResource.java:43]: Thought so too, was not sure. Will change it.

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
thanks a lot @atomfrede !

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/73020

**Comment by pascalgrimaud** [comment]
@atomfrede : approved

**Pull request merged**